guides/git-workflow.md

```diff
@@ -1,21 +1,89 @@
+
+## Which workflow should I use? <a name="which-workflow"></a>
+There are various pros and cons to each workflow, and it may not be obvious as to which workflow is best to choose for your project.  However, the power and benefit of the workflow comes in its consistent use across all team members and collaborators during the development of the software.
+
+### Branching Workflow <a name="branching-pros-cons"></a>
+
+**Pros:**
+- Continuous integration tools such as Travis or Jenkins can be easily enabled for all branches of the repository, so new feature branches can automatically be tested when a pull request is opened or modified.
+
+**Cons:**
+- It is not possible for external collaborators to contribute unless they explicitly have write access to the `spacetelescope` repository.
+- It is easier to accidentally delete or overwrite the work of others who are also working on the same branch, or happen to have their own branch of the same name.  While it is possible to recover this work or to restrict the editing of specific branches to only select developers, it can sometimes be a painstaking process.
+- Repositories may accrue a fair amount of stale branches over time, which may become burdensome for developers, and may cause collaborators to infer that the project is not being maintained.
+
+
+### Forking Workflow <a name="forking-pros-cons"></a>
+
+**Pros:**
+- Promotes "open development", wherein potential external collaborators who are interested in contributing may do so despite not having write access to the `spacetelescope` repository.
+- Provides a consistent experience and barrier-to-entry for everyone involved in the project, whether you are an internal or external collaborator.
+- Allows for an explicit namespace in pull requests and repository history that makes it more apparent who "owns" which branch.
+- Allows for contributors to experiment with various repository settings and/or continuous integrations tools without affecting how the `spacetelescope` repository operates.
+
+**Cons:**
+- Requires extra work to keep up-to-date with the parent, 'upstream' repository (i.e. one must periodically "update" their personal fork).
+- If forks are of a public repository, and more than one person wants to collaborate on someone's personal fork, then one must add each colleague to your fork.
```

Comment:
This is (in my opinion) a main motivating use case for the branching workflow instead. If you have a branch that you want multiple people to collaborate on, then it's useful for that branch to be in the shared organizational repository, rather than a branch in someone's fork. The forking into each developer's personal repo model works well for the use case of "one developer per feature (per fork or per branch)". That's true often, but not always...

Reply:
I still don't see the problem. One can PR (from one's own fork) to an existing PR (from another's fork). Slightly more complicated, yes, but not impossible.

Also, if you are a maintainer of the main repo, you can also push directly to someone else's branch that is tied to the PR, even if the PR is opened from a fork.

Reply:
I think the point I was trying to make here is that, if someone would like to push changes directly to someone else's personal fork, the pusher does not inherently have write access and must be added as a collaborator to the fork.

So perhaps I can just change this to:
>If a person wants to collaborate on a someone's personal fork directly, the one must be explicitly added as a collaborator to that fork.

?

Reply:
> If a person wants to collaborate on a someone's personal fork directly, the one must be explicitly added as a collaborator to that fork.

That is only necessary if that person is *not* a maintainer of upstream repo of which a PR is opened from that branch.